internal/cmd/ske/kubeconfig/create/create.go

```diff
@@ -0,0 +1,198 @@
+				"$ stackit ske kubeconfig create my-cluster --expiration 30d"),
+			examples.NewExample(
+				`Create a kubeconfig for the SKE cluster with name "my-cluster" and set the expiration time to 2 months`,
+				"$ stackit ske kubeconfig create my-cluster --expiration 2M"),
+			examples.NewExample(
+				`Create a kubeconfig for the SKE cluster with name "my-cluster" in a custom location`,
+				"$ stackit ske kubeconfig create my-cluster --location /path/to/config"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to create a kubeconfig for SKE cluster %q? This will OVERWRITE your current configuration, if it exists.", model.ClusterName)
+				err = confirm.PromptForConfirmation(cmd, prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create kubeconfig for SKE cluster: %w", err)
+			}
+
+			// Create a config file in $HOME/.kube/config
+			configPath := model.KubeconfigPath
+
+			if configPath == nil {
+				userHome, err := os.UserHomeDir()
+				if err != nil {
+					return fmt.Errorf("could not get user home directory: %w", err)
+				}
+
+				err = os.MkdirAll(fmt.Sprintf("%s/.kube", userHome), 0o700)
+				if err != nil {
+					return fmt.Errorf("could not create kube directory: %w", err)
+				}
+				configPath = utils.Ptr(fmt.Sprintf("%s/.kube", userHome))
+			}
+
+			err = os.WriteFile(fmt.Sprintf("%s/config", *configPath), []byte(*resp.Kubeconfig), 0o600)
+			if err != nil {
+				return fmt.Errorf("could not write kubeconfig file: %w", err)
+			}
+
+			fmt.Printf("Created kubeconfig file for cluster %s with expiration date %v.\n", model.ClusterName, *resp.ExpirationTimestamp)
+
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP(expirationFlag, "e", "", "Expiration time for the kubeconfig in seconds(s), minutes(m), hours(h), days(d) or months(M). Example: 30d. By default, expiration time is 1h")
+	cmd.Flags().String(locationFlag, "", "Folder location to store the kubeconfig file. By default, the kubeconfig is created in the .kube folder, in the user's home directory.")
+}
+
+func parseInput(cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	clusterName := inputArgs[0]
+
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	expirationTimeInput := flags.FlagToStringPointer(cmd, expirationFlag)
+	var expirationTime *string
+	if expirationTimeInput != nil {
+		expirationTime = convertToSeconds(*expirationTimeInput)
```

Comment:
Is there any other `parseInput` where we are adding extra logic besided mapping the flag to the model? If not I think we could separate these two and convert to seconds before/when building the request